app/src/main/java/com/kickstarter/viewmodels/SearchViewModel.java

```diff
@@ -24,93 +24,131 @@
+
+      final Observable<DiscoveryParams> searchParams = search
+        .filter(StringUtils::isPresent)
+        .debounce(300, TimeUnit.MILLISECONDS, scheduler)
+        .map(s -> DiscoveryParams.builder().term(s).build());
+
+      final Observable<DiscoveryParams> popularParams = search
+        .filter(StringUtils::isEmpty)
+        .map(__ -> defaultParams)
+        .startWith(defaultParams);
+
+      final Observable<DiscoveryParams> params = Observable.merge(searchParams, popularParams);
+
+      final ApiPaginator<Project, DiscoverEnvelope, DiscoveryParams> paginator =
+        ApiPaginator.<Project, DiscoverEnvelope, DiscoveryParams>builder()
+          .nextPage(nextPage)
+          .startOverWith(params)
+          .envelopeToListOfData(DiscoverEnvelope::projects)
+          .envelopeToMoreUrl(env -> env.urls().api().moreProjects())
+          .clearWhenStartingOver(true)
+          .concater(ListUtils::concatDistinct)
+          .loadWithParams(apiClient::fetchProjects)
+          .loadWithPaginationPath(apiClient::fetchProjects)
+          .build();
+
+      this.search
+        .filter(StringUtils::isEmpty)
+        .compose(bindToLifecycle())
+        .subscribe(__ -> searchProjects.onNext(ListUtils.empty()));
+
+      params
+        .compose(Transformers.takePairWhen(paginator.paginatedData()))
+        .compose(bindToLifecycle())
+        .subscribe(paramsAndProjects -> {
+          if (paramsAndProjects.first.sort() == defaultSort) {
+            popularProjects.onNext(paramsAndProjects.second);
+          } else {
+            searchProjects.onNext(paramsAndProjects.second);
+          }
+        });
+
+      // Track us viewing this page
+      this.koala.trackSearchView();
+
+      // Track search results and pagination
+      final Observable<Integer> pageCount = paginator.loadingPage();
+      final Observable<String> query = params
+        .map(DiscoveryParams::term);
+
+      query
+        .compose(Transformers.takePairWhen(pageCount))
+        .filter(qp -> StringUtils.isPresent(qp.first))
+        .compose(bindToLifecycle())
+        .subscribe(qp -> koala.trackSearchResults(qp.first, qp.second));
+    }
+
+    private static final DiscoveryParams.Sort defaultSort = DiscoveryParams.Sort.POPULAR;
+    private static final DiscoveryParams defaultParams = DiscoveryParams.builder().sort(defaultSort).build();
+
+    // Inputs
+    private final PublishSubject<Void> nextPage = PublishSubject.create();
+    private final PublishSubject<Project> projectCardClicked = PublishSubject.create();
+    private final PublishSubject<String> search = PublishSubject.create();
+
+    // Outputs
+    private final BehaviorSubject<List<Project>> popularProjects = BehaviorSubject.create();
+    private final BehaviorSubject<List<Project>> searchProjects = BehaviorSubject.create();
+
+    public final SearchViewModel.Inputs inputs = this;
+    public final SearchViewModel.Outputs outputs = this;
+
+    // Inputs
+    @Override
```

Comment:
you can go ahead and put the override on the same line, for all of these input and output functions i.e.

```java
@Override public void nextPage() { ... }
```